src/core/TChem_Driver.cpp

```diff
@@ -0,0 +1,279 @@
+  using kinetic_model_host_type = TChem::KineticModelNCAR_ConstData<host_device_type>;
+
+  using ordinal_type = TChem::ordinal_type;
+
+  g_tchem->createGasKineticModel(chemFile);
+  g_tchem->createGasKineticModelConstData();
+  g_tchem->createStateVector();
+  g_tchem->getLengthOfStateVector();
+  g_tchem->createNumerics(numericsFile);
+
+}
+
+/* Finalize the model by freeing memory and finalizing Kokkos */
+void finalize(){
+  g_tchem->freeAll();
+  delete g_tchem;
+  Kokkos::finalize();
+}
+
+/* Read in the solver settings */
+void TChem::Driver::createNumerics(const std::string &numerics_file) {
+
+   YAML::Node root = YAML::LoadFile(numerics_file);
+
+   auto atol_newton = root["solver_info"]["atol_newton"];
+   auto rtol_newton =  root["solver_info"]["rtol_newton"];
+   auto dtmin = root["solver_info"]["dtmin"];
+   auto atol_time = root["solver_info"]["atol_time"];
+   auto rtol_time = root["solver_info"]["rtol_time"];
+   auto max_num_newton_iterations = root["solver_info"]["max_newton_iterations"];
+   auto max_num_time_iterations = root["solver_info"]["max_time_iterations"];
+   auto num_time_iterations_per_interval = root["solver_info"]["max_time_iterations"];
+   auto jacobian_interval = root["solver_info"]["jacobian_interval"];
+
+   _atol_newton = atol_newton.as<real_type>(1e-10);
+   _rtol_newton = rtol_newton.as<real_type>(1e-6);
+   _dtmin = dtmin.as<real_type>(1e-8);
+   _atol_time = atol_time.as<real_type>(1e-12);
+   _rtol_time = rtol_time.as<real_type>(1e-4);
+   _max_num_newton_iterations = max_num_newton_iterations.as<ordinal_type>(100);
+   _max_num_time_iterations = max_num_time_iterations.as<ordinal_type>(1e3);
+   _num_time_iterations_per_interval = num_time_iterations_per_interval.as<ordinal_type>(1e3);
+   _jacobian_interval = jacobian_interval.as<ordinal_type>(1);
+
+}
+
+/* Create the gas kinetic model from a YAML file */
+void TChem::Driver::createGasKineticModel(const std::string &chem_file) {
+  _chem_file = chem_file;
+  _kmd = KineticModelData(_chem_file);
+}
+
+/* Create the gas kinetic constant model data */
+void TChem::Driver::createGasKineticModelConstData() {
+  printf("Creating kmcd \n");
+  using interf_host_device_type = typename Tines::UseThisDevice<TChem::host_exec_space>::type;
+  _kmcd_host = TChem::createNCAR_KineticModelConstData<interf_host_device_type>(_kmd);
+  _kmcd_device = TChem::createNCAR_KineticModelConstData<device_type>(_kmd);
+  printf("Number of Species %d \n", _kmcd_host.nSpec);
+  printf("Number of Reactions %d \n", _kmcd_host.nReac);
+  printf("End creating kmcd \n");
+
+}
+
+/* Free gas kinetic model */
+void TChem::Driver::freeAll() {
+  g_tchem->freeGasKineticModel();
+}
+
+void TChem::Driver::freeGasKineticModel() {
+  _chem_file = std::string();
+  _therm_file = std::string();
+}
+
+void TChem::Driver::createStateVector() {
+  const ordinal_type len = TChem::Impl::getStateVectorSize(_kmcd_host.nSpec);
+  const ordinal_type nBatch = 1;
```

Comment:
Should we pass nBatch as an input?

Reply:
Added as argument in [3c61a7c](https://github.com/PCLAeroParams/TChem-atm/pull/55/commits/3c61a7c28bf1d734db5fd5581c2020267aac6af7)